mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt

```diff
@@ -8,127 +8,48 @@ import android.view.View
+    private val viewModel: BankAccountsViewModel by viewModels()
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        val rootView = inflater.inflate(R.layout.fragment_accounts, container, false)
-        ButterKnife.bind(this, rootView)
-        setupRecycletView()
-        setUpSwipeRefresh()
-        mPresenter!!.attachView(this)
-        showSwipeProgress()
-        mBankAccountsPresenter!!.fetchLinkedBankAccounts()
-        return rootView
-    }
-
-    private fun setUpSwipeRefresh() {
-        swipeRefreshLayout?.setOnRefreshListener { mBankAccountsPresenter!!.fetchLinkedBankAccounts() }
-    }
-
-    private fun setupRecycletView() {
-        val layoutManager = LinearLayoutManager(context)
-        layoutManager.orientation = LinearLayoutManager.VERTICAL
-        mRvLinkedBankAccounts!!.layoutManager = layoutManager
-        mRvLinkedBankAccounts!!.setHasFixedSize(true)
-        mRvLinkedBankAccounts!!.adapter = mBankAccountsAdapter
-        mRvLinkedBankAccounts!!.addItemDecoration(
-            DividerItemDecoration(
-                context,
-                DividerItemDecoration.VERTICAL
-            )
-        )
-        mRvLinkedBankAccounts!!.addOnItemTouchListener(
-            RecyclerItemClickListener(
-                activity,
-                object : RecyclerItemClickListener.OnItemClickListener {
-                    override fun onItemClick(childView: View?, position: Int) {
-                        val intent = Intent(
-                            activity,
-                            BankAccountDetailActivity::class.java
-                        )
-                        intent.putExtra(
-                            Constants.BANK_ACCOUNT_DETAILS,
-                            mBankAccountsAdapter!!.getBankDetails(position)
-                        )
-                        intent.putExtra(Constants.INDEX, position)
-                        startActivityForResult(intent, BANK_ACCOUNT_DETAILS_REQUEST_CODE)
-                    }
-
-                    override fun onItemLongPress(childView: View?, position: Int) {}
-                })
-        )
-    }
-
-    override fun showLinkedBankAccounts(bankAccountList: List<BankAccountDetails?>?) {
-        if (bankAccountList == null || bankAccountList.size == 0) {
-            mRvLinkedBankAccounts!!.visibility = View.GONE
-            linkedAccountsText!!.visibility = View.GONE
-            setupUi()
-        } else {
-            hideEmptyStateView()
-            mRvLinkedBankAccounts!!.visibility = View.VISIBLE
-            linkedAccountsText!!.visibility = View.VISIBLE
-            mBankAccountsAdapter!!.setData(bankAccountList)
+        binding= FragmentAccountsBinding.inflate(inflater, container, false)
+        ButterKnife.bind(this,binding.root)
+        binding.accountScreen.setContent {
+            AccountScreen(viewModel, ::addAccountClicked, ::onAccountClicked)
```

Comment:
Don't need to pass the viewmodel init the ViewModel through hilt. 

See `CardsScreen` as an example.

Reply:
done